apps/server/src/modules/user/legacy/controller/api-test/admin-api-teachers.api.spec.ts

```diff
@@ -0,0 +1,258 @@
+		em.persist(school);
+		em.persist(currentYear);
+		em.persist([adminRoles, teacherRoles]);
+		em.persist([adminUser, teacherUser1, teacherUser2]);
+		em.persist([adminAccount, teacherAccount1, teacherAccount2]);
+		await em.flush();
+	};
+
+	beforeAll(async () => {
+		const moduleFixture: TestingModule = await Test.createTestingModule({
+			imports: [ServerTestModule],
+		})
+			.overrideGuard(JwtAuthGuard)
+			.useValue({
+				canActivate(context: ExecutionContext) {
+					const req: Request = context.switchToHttp().getRequest();
+					req.user = currentUser;
+					return true;
+				},
+			})
+			.compile();
+
+		app = moduleFixture.createNestApplication();
+		await app.init();
+		em = app.get(EntityManager);
+
+		await setupDb();
+	});
+
+	afterAll(async () => {
+		await app.close();
+		em.clear();
+	});
+
+	describe('[GET] :id', () => {
+		describe('when teacher exists', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+			};
+
+			it('should return teacher ', async () => {
+				setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}/${teacherUser1.id}`)
+					.expect(200);
+
+				// eslint-disable-next-line @typescript-eslint/naming-convention
+				const { _id } = response.body as UserResponse;
+
+				expect(_id).toBe(teacherUser1._id.toString());
+			});
+		});
+
+		describe('when user has no right permission', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(teacherUser1, teacherAccount1);
+			};
+
+			it('should reject request', async () => {
+				setup();
+				await request(app.getHttpServer()) //
+					.get(`${basePath}/${teacherUser1.id}`)
+					.expect(403);
+			});
+		});
+
+		describe('when teacher does not exists', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+			};
+
+			it('should reject request ', async () => {
+				setup();
+				await request(app.getHttpServer()) //
+					.get(`${basePath}/000000000000000000000000`)
+					.expect(404);
+			});
+		});
+	});
+
+	describe('[GET]', () => {
+		describe('when sort param is provided', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+				const query: UsersSearchQueryParams = {
+					$skip: 0,
+					$limit: 5,
+					$sort: { firstName: 1 },
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return teachers in correct order', async () => {
+				const { query } = setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}`)
+					.query(query)
+					.set('Accept', 'application/json')
+					.expect(200);
+
+				const { data, total } = response.body as UserListResponse;
+
+				expect(total).toBe(2);
+				expect(data.length).toBe(2);
+				expect(data[0]._id).toBe(teacherUser1._id.toString());
+				expect(data[1]._id).toBe(teacherUser2._id.toString());
+			});
+		});
+
+		describe('when search params are too tight', () => {
```

Comment:
I think you can put this test also in the repo test. Then you don't have to duplicate it here and on the admin-api-students tests.